### PR TITLE
234 blockgrid functional and e2e test

### PR DESCRIPTION
**Pull request by LarryAmisola**
**Explain the _details_ for making this change. What existing problem does the pull request solve?**
Add API and E2E Test for BlockGrid Component.

**Related github/jira issue (required)**:
#234 

**Steps necessary to review your pull request (required)**:
- Run the tests:
    ```sh
    npm run functional:ci
    npm quickstart # or "start" so the app is running
    npm run e2e:ci
    ```

### CODE REVIEW

File: test/components/blockgrid/blockgrid.e2e-spec.js [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    browser.actions().mouseMove(blockEl1).click().perform();
+    browser.actions().mouseMove(checkEl1).click().perform();
+
+    browser.actions().mouseMove(blockEl2).click().perform();
+    browser.actions().mouseMove(checkEl2).click().perform();
+
+    expect(await element(by.className('is-selected'))).toBeTruthy();
+  });
+});
+
```
Comment (LarryAmisola) [inline test/components/blockgrid/blockgrid.e2e-spec.js]: Not sure why I got this message when running e2e:ci at the end of this block
'StaleElementReferenceError: stale element reference: element is not attached to the page document'

Comment (tmcconechy) [reply test/components/blockgrid/blockgrid.e2e-spec.js]: Make sure to put an await on every line to get rid of that...See notes.

File: test/components/blockgrid/blockgrid.e2e-spec.js [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+    browser.actions().mouseMove(blockEl1).click().perform();
+    browser.actions().mouseMove(blockEl2).click().perform();
+
+    expect(await element(by.className('is-activated'))).toBeTruthy();
+  });
+
+  it('Should blocks checked after selected', async () => {
+    const blockEl1 = await element.all(by.css('.block.is-selectable')).get(1);
+    const blockEl2 = await element.all(by.css('.block.is-selectable')).get(2);
+    const checkEl1 = element(by.css("label[for='checkbox0']"));
```
Comment (tmcconechy) [inline test/components/blockgrid/blockgrid.e2e-spec.js]: await

File: test/components/blockgrid/blockgrid.e2e-spec.js [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+    const checkEl2 = element(by.css("label[for='checkbox1']"));
```
Comment (tmcconechy) [inline test/components/blockgrid/blockgrid.e2e-spec.js]: await

File: test/components/blockgrid/blockgrid.e2e-spec.js [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+
+    browser.actions().mouseMove(blockEl1).click().perform();
```
Comment (tmcconechy) [inline test/components/blockgrid/blockgrid.e2e-spec.js]: await

File: test/components/blockgrid/blockgrid.e2e-spec.js [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+  });
+
+  it('Should singleselect blocks be selectable', async () => {
+    expect(await element.all(by.className('is-selectable'))).toBeTruthy();
+  });
+
+  it('Should blocks singleselect highlight after clicked', async () => {
+    const blockEl1 = await element.all(by.css('.block.is-selectable')).get(1);
+
+    browser.actions().mouseMove(blockEl1).click().perform();
```
Comment (tmcconechy) [inline test/components/blockgrid/blockgrid.e2e-spec.js]: await

Comment (LarryAmisola) [reply test/components/blockgrid/blockgrid.e2e-spec.js]: Noted Tim

**Comment by tmcconechy** [comment]
@CindyMercadoReyes please review these tests

**Pull request merged**